pkg/controllers/user/nodesyncer/nodessyncer.go

```diff
@@ -364,6 +364,17 @@ func (m *NodesSyncer) convertNodeToNode(node *corev1.Node, existing *v3.Node, po
 	for name, quantity := range limits {
 		machine.Status.Limits[name] = quantity
 	}
+	if node.Labels != nil {
+		if _, ok := node.Labels["node-role.kubernetes.io/etcd"]; ok {
+			machine.Spec.Etcd = true
+		}
+		if _, ok := node.Labels["node-role.kubernetes.io/controlplane"]; ok {
```

Comment:
Do you think we could look for master here to?  Isn't that what most other k8s distro use?